### PR TITLE
Add caller capability

### PR DESCRIPTION
**Pull request by omidnikta**
This pull request adds the capability of adding file and line number to the logger. Use
ShowCaller(true/false) to enable/disable it.

### CODE REVIEW

File: entry_test.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func TestEntryWithError(t *testing.T) {
 
 	logger := New()
 	logger.Out = &bytes.Buffer{}
-	entry := NewEntry(logger)
+	entry := NewEntry(logger, 1)
```
Comment (aybabtme) [inline entry_test.go]: the test should assert that the right caller is printed

File: entry.go [hunk 2]
```diff
@@ -29,16 +31,23 @@ type Entry struct {
...
 		Logger: logger,
 		// Default is three fields, give a little extra room
-		Data: make(Fields, 5),
+		Data:  make(Fields, 5),
+		Depth: d,
 	}
 }
 
+func (entry *Entry) deepen() {
+	entry.Depth = entry.Depth + 1
```
Comment (aybabtme) [inline entry.go]: this mutates the entry, which means multiple calls will result in bogus callers

Comment (omidnikta) [inline entry.go]: As far as I know, when you call to log, a new entry is created. So no data race is possible.

File: entry_test.go (hunk 1 above)
Comment (omidnikta) [inline entry_test.go]: There is no NewEntry(logger) anymore. You need to feed a number to it like NewEntry(logger, 1). When you set SetCaller(false), the caller will have no effect.

File: entry.go (hunk 2 above)
Comment (aybabtme) [inline entry.go]: nha, `Entry` are reusable

File: entry_test.go (hunk 1 above)
Comment (aybabtme) [inline entry_test.go]: what I'm saying is that the test assertions need to check that the right `callrer: src.go:42` is printed

File: entry.go (hunk 2 above)
Comment (omidnikta) [inline entry.go]: Latest commit fixes the reusing entry bug, if an entry will not be used concurrently.

File: entry_test.go (hunk 1 above)
Comment (omidnikta) [inline entry_test.go]: This problem is now solved.

File: entry.go (hunk 2 above)
Comment (omidnikta) [inline entry.go]: This problem is solved now.

**Comment by aybabtme** [comment]
closing due to inactivity and failed CI.

**Comment by apriendeau** [comment]
@aybabtme this failed due to travis. WE should consider reopening:

```
Cloning into '/home/travis/gopath/src/github.com/logrus/logrus'...
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/logrus/logrus/'
package github.com/logrus/logrus: exit status 128
```

**Comment by aybabtme** [comment]
@apriendeau travis failed because the code is incorrect.

**Comment by apriendeau** [comment]
Well the logs are very vague and not useful. @aybabtme are we okay with the implementation because I will fix the import paths if we are.

**Comment by aybabtme** [comment]
@apriendeau the proposed change in this PR breaks existing users and is larger than strictly necessary. I'm not personally in favor of merging it.

**Pull request closed without merging**